source/libraries/CompileSolidity.ts

```diff
@@ -11,58 +11,66 @@ interface CompileContractsOutput {
     output?: string;
 }
 
-class SolidityContractCompiler {
+export class SolidityContractCompiler {
     private contractInputDirectoryPath: string;
     private contractOutputDirectoryPath: string;
     private contractOutputFileName: string;
 
     public constructor(contractInputDirectoryPath: string, contractOutputDirectoryPath: string, contractOutputFileName: string) {
+        if (contractInputDirectoryPath.lastIndexOf(path.sep) != contractInputDirectoryPath.length - 1) {
+            contractInputDirectoryPath += path.sep;
+        }
+        if (contractOutputDirectoryPath.lastIndexOf(path.sep) != contractOutputDirectoryPath.length - 1) {
+            contractOutputDirectoryPath += path.sep;
+        }
         this.contractInputDirectoryPath = contractInputDirectoryPath;
         this.contractOutputDirectoryPath = contractOutputDirectoryPath;
         this.contractOutputFileName = contractOutputFileName;
     }
 
-    public async compileContracts(): Promise<CompileContractsOutput> {
+    public readCallback(path: string): { contents?: string, error?: string } {
         try {
-            // Compile all contracts in the specified input directory
-            const compilerInputJson = await this.generateCompilerInput();
-            const compilerOutputJson: string = compileStandardWrapper(JSON.stringify(compilerInputJson), this.readCallback);
-            const compilerOutput: CompilerOutput = JSON.parse(compilerOutputJson);
-            if (compilerOutput.errors) {
-                let errors = "";
-                for (let error of compilerOutput.errors) {
-                    errors += error.formattedMessage + "\n";
-                }
-                throw new Error("The following errors/warnings were returned by solc:\n\n" + errors);
-            }
+            const result = fs.readFileSync(path, 'utf8');
```

Comment:
We should have read in all files by the time this is called so we should be able to just reference the previously-read file here.  Alternatively, maybe we should just remove the `readCallback` and let things crash if some file is attempted to be read that doesn't exist?

Reply:
Yeah, I suppose we don't really need `readCallback` here.  I'll just remove it.